### PR TITLE
postprocess commonjs and amd outputs to make them easier to consume

### LINKED ISSUE #10461: AMD bundling: strip out process.env.NODE_ENV which isn't automatically defined by bundling
https://github.com/OfficeDev/office-ui-fabric-react/commit/a9245feee547762f0c8c9935cbf39d737b86c519 clearly introduces a breaking change without bumping the major version.

If fabric folks intend to use the minor versions as breaking, then please add guidelines that consumers should be fixing their fabric versions precisely up to the patch version.

### PR DESCRIPTION
**Pull request by kenotron**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #10461
- [x] Include a change request file using `$ yarn change`

#### Description of changes

This introduces some postprocessing code for the AMD and COMMONJS module types from Fabric:

1. AMD users do not have a `process.env` defined - this is a node concept and is only declared by webpack
2. COMMONJS users who use jest to test their code have had to transform OUFR code because even if they required the 'lib-commonjs' version of our code, we still internally reference /lib everywhere (commonjs requiring esm) which is incompatible. 

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10561)

### CODE REVIEW

**Comment by dzearing** [comment]
Thanks @kenotron !!!

**Comment by kenotron** [comment]
You’re welcome! I’m not quite ready to merge just yet... need to think through implications and need someone who use AMD to help verify this...

File: scripts/tasks/postprocess-commonjs.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  const getAllPackageInfo = require('../monorepo/getAllPackageInfo');
+  const allPackages = Object.keys(getAllPackageInfo());
+
+  mod('lib-commonjs/**/*.js').asTypescript((node, modder) => {
+    if (ts.isCallExpression(node)) {
+      if (node.expression.getText() === 'require' && node.arguments.length === 1) {
+        const arg = node.arguments[0];
+        if (ts.isStringLiteral(arg)) {
+          if (belongsToFabric(arg.text) && arg.text.includes('lib/')) {
+            modder.replace(arg, `"${arg.text.replace('lib', 'lib-commonjs')}"`);
```
Comment (dzearing) [inline scripts/tasks/postprocess-commonjs.js]: This should really include the trailing slash so there are no accidents :)

File: scripts/tasks/postprocess-commonjs.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          }
+        }
+      }
+    }
+  });
+
+  function belongsToFabric(requireArg) {
```
Comment (dzearing) [inline scripts/tasks/postprocess-commonjs.js]: I don't understand what this is doing or how it relates to fabric. Prefer no "fabricisms" in this script if possible :)

Comment (dzearing) [reply scripts/tasks/postprocess-commonjs.js]: I think this is filtering out imports that are in the local repo

Comment (kenotron) [reply scripts/tasks/postprocess-commonjs.js]: Well, this should only convert lib-commonjs for fabric related packages so inherently fabric related. This function picks up all the local packages as something we would postprocess for a file.

**Comment by kenotron** [comment]
I disabled the commonjs postprocess for now just so we can progress with the fix for the AMD post processing. That commonjs postprocess would be turned on in another isolated change.

**Comment by kenotron** [comment]
Got an offline approval from @dzearing, so I'm merging this one.

**Pull request merged**